etcdserver/cluster_util.go

```diff
@@ -252,3 +252,12 @@ func getVersion(m *Member, tr *http.Transport) (*version.Versions, error) {
 	}
 	return nil, err
 }
+
+func MustDetectDowngrade(cv *semver.Version) {
+	lv := semver.Must(semver.NewVersion(version.Version))
+	// only keep major.minor version for comparison against cluster version
+	lv = &semver.Version{Major: lv.Major, Minor: lv.Minor}
+	if cv != nil && lv.LessThan(*cv) {
+		plog.Fatalf("cluster cannot be downgraded. (previous cluster version: %s, current version: %s)", version.Cluster(cv.String()), version.Cluster(lv.String()))
```

Comment:
ok. always use binary version.

The current output is `cluster cannot be downgraded. (previous cluster version: 2.2, current version: 2.1)`, and i think `cluster cannot be downgraded. (previous cluster version: 2.2, current binary version: 2.1.1)` is better. Users may have question about what current version stands for, especially when it is not a standard semver.